model/task_history.go

```diff
@@ -364,8 +364,9 @@ func (self *taskHistoryIterator) GetDistinctTestNames(numCommits int) ([]string,
 		[]bson.M{
 			{
 				"$match": bson.M{
-					task.BuildVariantKey: bson.M{"$in": self.BuildVariants},
-					task.DisplayNameKey:  self.TaskName,
+					task.ProjectKey:     self.ProjectName,
```

Comment:
which index are you trying to use with this change? I see a branch_1_r_1_build_variant_1 index which should work very well if you add the variant filter back into this one

Reply:
I wanted to avoid having to do a variant query, and I believe $in is not supposed to have as good of performance. There's a branch_1_r_1_display_name_1_create_time_1 index I thought would work, let me reorder though.

Reply:
The ordering doesn't matter, both because bson.M isn't ordered and also the query optimizer will look at all filter criteria at once. That index should have a higher cardinality than the one I suggested though so that works